### PR TITLE
Mpi emigration collision checking

### LINKED ISSUE #145: In MPI emigration check particle conjunctions before leaving rank (theoretically emigration can conincide with upcoming detection)
# Issue
## The problem
As title
## Expected behavior
Not as title
## What needs to be done
- [ ] Check specifically on emigrants
- [ ] Add test to check this keeps working
## How it can be tested or reproduced
Test case

### PR DESCRIPTION
**Pull request by gomezzz**
# Description

- Added collision checks on the rank emigrants 
- Minor formatting and naming
- Fixed bug in collisionDetection around particles (endTraversal was missing leading to no collision being detected)
- Added setters for safe particle ids to aid tests (otherwise ids are not rankspecific in tests)
- Renamed regular grid decomp tests
- Added test for collision detection amongst leaving and incoming particles on MPI ranks
- Added checking for collision between currently migrating particles

## Related Pull Requests

- #138 

## Resolved Issues

- [x] fixes #145 

## How Has This Been Tested?

New tests, etc.

### CODE REVIEW

**Comment by gomezzz** [comment]
@FG-TUM There is a problem with regular grid decomposition. Somehow, the particle communication is not correct and the particles moving from one rank to the other don't arrive. The test is currently failing for that. I don't really wanna spend time on this as we aren't using it right now anyways. Any suggestions?

**Comment by gomezzz** [comment]
Tests are fixed in #151

File: src/ladds/distributedMemParallelization/ParticleMigrationHandler.h [hunk 1]
```diff
@@ -35,14 +35,16 @@ namespace LADDS::ParticleMigrationHandler {
...
-LADDS::CollisionFunctor::CollisionCollectionT collisionDetectionImmigrants(
+LADDS::CollisionFunctor::CollisionCollectionT collisionDetectionAroundParticles(
     AutoPas_t &autopas,
     std::vector<LADDS::Particle> &incomingParticles,
     double deltaT,
     double maxV,
     double collisionDistanceFactor,
-    double minDetectionRadius);
+    double minDetectionRadius,
+    bool checkForInternalCollisions = false);
```
Comment (FG-TUM) [inline src/ladds/distributedMemParallelization/ParticleMigrationHandler.h:48]: It's less error-prone to not use a default argument

File: tests/mpitestladds/SimulationTest.cpp [hunk 2]
```diff
@@ -1,131 +0,0 @@
-/**
- * @file SimulationTest.cpp
```
Comment (FG-TUM) [inline tests/mpitestladds/SimulationTest.cpp:2]: was this just deleted or moved?

File: tests/mpitestladds/distributedMemParallelization/RegularGridDecompositionTests.cpp [hunk 3]
```diff
@@ -0,0 +1,273 @@
+/**
+ * @file RegularGridDecompositionTests.cpp
```
Comment (FG-TUM) [inline tests/mpitestladds/distributedMemParallelization/RegularGridDecompositionTests.cpp:2]: seems to have some overlap with AltitudeBasedDecompositionTests.cpp. Could probably benefit from some common super class for setup stuff.

**Review by FG-TUM — CHANGES_REQUESTED** [review]

**Comment by FG-TUM** [comment]
> @FG-TUM There is a problem with regular grid decomposition. Somehow, the particle communication is not correct and the particles moving from one rank to the other don't arrive. The test is currently failing for that. I don't really wanna spend time on this as we aren't using it right now anyways. Any suggestions?

Either we fix it or we add a warning + issue when using it

File: tests/mpitestladds/SimulationTest.cpp (hunk 2 above)
Comment (gomezzz) [reply tests/mpitestladds/SimulationTest.cpp:2]: didn't use git move

**Comment by FG-TUM** [comment]
seems like the tests take forever now?

**Comment by gomezzz** [comment]
> seems like the tests take forever now?

I think this happens if they fail, not sure why they don't just fail? If they pass however, they pass normally

**Pull request merged**